### PR TITLE
For #1462: Introduce text.Mapped

### PR DESCRIPTION
**Pull request by andreoss**
Per #1462 
- Add `text.Mapped`
- Use it where applicable

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1485#issuecomment-710775144))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1485](https://github.com/yegor256/cactoos/issues/1485) is [in scope](http://www.zerocracy.com/policy.html#14) for 18hr
  * The role is `REV`
  * The job is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for [17hr](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @andreoss[/z](https://www.0crat.com/u/andreoss): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231485%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by andreoss** [comment]
@fabriciofx Please take a look

File: src/test/java/org/cactoos/text/MappedTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void mapsWithFormat() {
+        new Assertion<>(
+            "must apply lambda to a string",
+            new Mapped(
+                s -> String.format("<%s>", s),
+                new TextOf("hi")
+            ),
+            new TextHasString("<hi>")
```
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java]: @fabriciofx Done. Why do we have `TextHasString`?

File: src/test/java/org/cactoos/text/MappedTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void maps() {
+        new Assertion<>(
+            "must apply mehtod reference to a string",
+            new Mapped(
+                String::toLowerCase,
+                new TextOf("ABC")
+            ),
+            new TextHasString("abc")
```
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java]: @fabriciofx Done.

File: src/test/java/org/cactoos/text/MappedTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                new TextOf("hi")
+            ),
+            new TextHasString("<hi>")
+        ).affirm();
+    }
+
+    @Test
+    void maps() {
+        new Assertion<>(
+            "must apply mehtod reference to a string",
```
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java]: @fabriciofx Done.

File: src/test/java/org/cactoos/text/MappedTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.TextHasString;
+
+/**
+ * Tests for @{link Mapped}.
+ *
+ * @since 0.47
+ */
+final class MappedTest {
```
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java:36]: @fabriciofx Why do you think `Mapped` should preserve equality?

[the remark this answers is not in the document]
Comment (fabriciofx) [reply src/test/java/org/cactoos/text/MappedTest.java:35]: @andreoss Some code like it `new Mapped(String::toLowercase, new TextOf("ABC").equals("abc")` is valid? If yes, than `Mapped` should preserve equality, right?

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1485#issuecomment-721031825))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1485](https://github.com/yegor256/cactoos/issues/1485) is [in scope](http://www.zerocracy.com/policy.html#14) for 18days
  * The role is `REV`
  * The job is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for [18days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @andreoss[/z](https://www.0crat.com/u/andreoss): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231485%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: src/test/java/org/cactoos/text/MappedTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java:35]: @fabriciofx In order to do that, it has to extend `TextEnvelope` and use `TextOf`. Both seem to be unnecessary for this task.

[the remark this answers is not in the document]
Comment (fabriciofx) [reply src/test/java/org/cactoos/text/MappedTest.java:35]: @andreoss Ok, thanks for this information!

**Review by fabriciofx — APPROVED** [review]
@andreoss It seems fine to me. @victornoel WDYT? Can you merge it?

File: src/test/java/org/cactoos/text/MappedTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (victornoel) [reply src/test/java/org/cactoos/text/MappedTest.java:35]: @andreoss @fabriciofx actually all texts implementation should either implement equals themselves or indeed delegate to an implementation of text that does. So I think we need to do it either way here yes. Sale with tostring and hashcode

**Review by victornoel — CHANGES_REQUESTED** [review]
@andreoss see the one comment

File: src/test/java/org/cactoos/text/MappedTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreoss) [reply src/test/java/org/cactoos/text/MappedTest.java:36]: @victornoel Done. It uses `TextOf` now.

**Comment by andreoss** [comment]
@victornoel `equals`/`hashCode` are provided by `TextOf` now. Please, take a look

**Comment by victornoel** [comment]
@andreoss great!

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#1485](https://github.com/yegor256/cactoos/issues/1485), since the job is not in scope anymore

**Pull request merged**